### PR TITLE
Deprecate getMain/setMain in JavaExec task

### PR DESCRIPTION
**Pull request by rieske**
Those have been replaced with a mainClass property.

https://github.com/gradle/gradle/issues/15681

### CODE REVIEW

File: subprojects/core-api/src/main/java/org/gradle/process/JavaExecSpec.java [hunk 1]
```diff
@@ -70,7 +73,10 @@
      * @param main the fully qualified name of the main class to be executed.
      *
      * @return this
+     *
+     * @deprecated Use {@link #getMainClass()}.set(main) instead. This method will be removed in Gradle 8.0.
      */
+    @Deprecated
```
Comment (donat) [inline subprojects/core-api/src/main/java/org/gradle/process/JavaExecSpec.java:79]: Please add the `@ReplacedBy` annotation here too.

**Review by donat — CHANGES_REQUESTED** [review]
Just a tiny request, otherwise looks good.

File: subprojects/core-api/src/main/java/org/gradle/process/JavaExecSpec.java (hunk 1 above)
Comment (rieske) [reply subprojects/core-api/src/main/java/org/gradle/process/JavaExecSpec.java:79]: Done.

**Review by donat — APPROVED** [review]

**Comment by rieske** [comment]
@bot-gradle test and merge this, please.

**Comment by rieske** [comment]
@bot-gradle test and merge

**Comment by rieske** [comment]
@bot-gradle test and merge

**Comment by rieske** [comment]
@bot-gradle test and merge

File: subprojects/core-api/src/main/java/org/gradle/process/JavaExecSpec.java (hunk 1 above)
Comment (rieske) [reply subprojects/core-api/src/main/java/org/gradle/process/JavaExecSpec.java:79]: I had to remove this again as there is some validation in the pipeline that breaks the build when `@ReplacedBy` is put on a setter which tells to only add this annotation to getters.

**Pull request merged**